Add Docker configuration

mihvas opened the pull request:
Technologies Used:

Docker & Docker Compose: For containerization and orchestration of services.
PHP 8.2-fpm: Container for executing PHP code based on the official PHP image.
Nginx: Web server that handles incoming HTTP requests and forwards them to PHP-FPM.
PostgreSQL: Database service for the project.
Symfony Framework & Doctrine ORM: The base Symfony project integrated with Doctrine ORM for database interactions.

Project Structure:

/symfony_project

├── docker-compose.yml # Describes the services: php, nginx, db, including a volume for persistent database storage.

├── Dockerfile-php # Dockerfile for building the PHP container with necessary extensions and Composer.

├── docker
│ └── nginx
│ └── default.conf # Nginx configuration file for routing requests to Symfony (root: /var/www/html/public).

├── .env # Environment file with DATABASE_URL set to connect to the 'db' container.

├── ... (other Symfony project files)

mihvas committed:
Added Docker configuration

Added docker-compose.yml, Dockerfile-php, docker/nginx/default.conf

Rmgs123 commented:
Wow!

feip-krk commented:
Текст коммита следует начинать с глагола в повелительном наклонении, то есть правильно будет:

Add Docker configuration #1

Что можно прочитать как: "После добавления этого коммита будет **добавлена конфигурация докера**"

feip-krk commented on docker-compose.yml:
```diff
@@ -0,0 +1,42 @@
+    environment:
+      - POSTGRES_USER=symfony
+      - POSTGRES_PASSWORD=symfony
+      - POSTGRES_DB=symfony
```
Параметры базы данных также стоит вынести в env файл, потому что они не должна быть в репозитории - это небезопасно, и потому что между разными серверами настройки могут отличаться. Когда они в env их проще настраивать, они не будут перетираться при загрузке обновления приложения

feip-krk commented:
Дополнительное пожелание: ты упростишь себе жизнь, если сразу напишешь инструкции по разворачиванию приложения в файл README

feip-krk commented on .env:
```diff
@@ -18,3 +18,5 @@
 APP_SECRET=
 ###< symfony/framework-bundle ###
+
+DATABASE_URL="postgresql://symfony:symfony@db:5432/symfony?serverVersion=15&charset=utf8"
```
Эта ссылка содержит логин/пароль и базу данных, вынеси их в отдельные переменные и собери тут из шаблона:

`DATABASE_URL="postgresql://${DATABASE_USERNAME}:${DATABASE_PASSWORD} ...`

Значения DATABASE_USERNAME , DATABASE_PASSWORD , DATABASE_DATABASE заполнять надо в локальном файле .env.local , а здесь либо оставить пустыми или (лучше так):

`DATABASE_PASSWORD=CHANGE_ME`

feip-krk commented on Dockerfile-php:
```diff
@@ -0,0 +1,18 @@
+FROM php:8.2-fpm
```
Перемести этот фал в docker/php (не забудь изменить ссылки на него)

feip-krk commented:
Проект запустился, молодец, исправь проблемы с безопасностью и желательно напиши инструкцию по разворачиванию

feip-krk commented on Dockerfile-php:
```diff
@@ -0,0 +1,18 @@
+    && docker-php-ext-install pdo pdo_pgsql
+
+# Установка Composer (если нужно внутри контейнера)
+RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" \
```
Зачем тебе может потребоваться композер внутри контейнера?

mihvas committed:
Refactor structure of the project and increase security

Security issues have been fixed. All secret data has been transferred to ".env.local". Instructions for deploying the application have also been added.

mihvas commented:
@feip-krk Проверьте, пожалуйста.

feip-krk commented on README.md:
```diff
@@ -1 +1,57 @@
+- **Git** – for version control.
+- **Docker & Docker Compose** – for containerization.
+- **PHP & Composer** – (optional).
+- **Symfony CLI** – (optional).
```
php и symfony не надо, потому что они устанавливаются в контейнере

mihvas replied:
@feip-krk можно ли выполнить слияние? А по поводу php и Symfony - я указал, что это является опциональным для некоторых частных случаев.

mihvas commented:
@feip-krk можно ли выполнить слияние? А по поводу php и Symfony - я указал, что это является опциональным для некоторых частных случаев.
